src/main/java/org/datadog/jmxfetch/reporter/Reporter.java

```diff
@@ -88,7 +88,10 @@ public void sendMetrics(LinkedList<HashMap<String, Object>> metrics, String inst
                 long now = System.currentTimeMillis();
                 double rate = 1000 * (currentValue - oldValue) / (now - oldTs);
 
-                if (!Double.isNaN(rate) && !Double.isInfinite(rate)) {
+                boolean submit = (!Double.isNaN(rate) && !Double.isInfinite(rate));
+                submit &= (rate >= 0 | !canonicalRate);
+
+                if  (submit) {
                     sendMetricPoint(metricType, metricName, rate, tags);
                 }
```

Comment:
nit: I think we should log something when we don't submit a metric because `canonicalRate` is enabled, could ease troubleshooting missing rates.